includes/course-video/blocks/class-sensei-course-video-blocks-video-extension.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * File containing the Sensei_Course_Video_Blocks_Youtube_Extension class.
+ *
+ * @package sensei-lms
+ * @since 4.0.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Extends standard Embed block with YouTube specific functionality for video course progression
+ *
+ * @since 4.0.0
+ */
+class Sensei_Course_Video_Blocks_Video_Extension {
```

Comment:
These classes also look very similar with many code duplications. I think we could create another class with all the common parts, and just extends that in the custom ones, so we could improve the maintainability. :)

I think the super class could have the methods for:

- Instance
- Init
- Construct
- Enqueue (if possible, enqueueing regardless the hook)
- Part of the wrap. It could call a specific method of each subclass for the custom stuff, like a check if it should add the wrapper? And the type could be an attribute?)